static/app/components/dropdownLink.spec.tsx

```diff
@@ -0,0 +1,314 @@
+        userEvent.click(screen.getByTestId('outside-element'));
+
+        jest.runAllTimers();
+
+        // Should still be open
+        expect(screen.getByText('hi')).toBeInTheDocument();
+      });
+
+      it('does not close when dropdown actor is clicked', function () {
+        render(
+          <DropdownLink title="test" alwaysRenderMenu={false} isOpen>
+            <li>hi</li>
+          </DropdownLink>
+        );
+
+        // Click menu
+        userEvent.click(screen.getByText('test'));
+
+        // Should still be open
+        expect(screen.getByText('hi')).toBeInTheDocument();
+      });
+    });
+    describe('Closed', function () {
+      it('does not open when dropdown actor is clicked', function () {
+        render(
+          <DropdownLink title="test" alwaysRenderMenu={false} isOpen={false}>
+            <li>hi</li>
+          </DropdownLink>
+        );
+
+        // Click menu
+        userEvent.click(screen.getByText('test'));
+
+        expect(screen.queryByText('hi')).not.toBeInTheDocument();
+      });
+    });
+  });
+
+  describe('Nested Dropdown', function () {
+    const NestedDropdown = () => {
+      return (
+        <DropdownLink title="parent" alwaysRenderMenu={false}>
+          <li>
+            <DropdownLink alwaysRenderMenu={false} title="nested" isNestedDropdown>
+              <li>
+                <DropdownLink alwaysRenderMenu={false} title="nested #2" isNestedDropdown>
+                  <li>Hello</li>
+                </DropdownLink>
+              </li>
+            </DropdownLink>
+          </li>
+          <li id="no-nest">Item 2</li>
+        </DropdownLink>
+      );
+    };
+
+    it('closes when top-level actor is clicked', function () {
+      render(<NestedDropdown />);
+
+      // Open menu
+      userEvent.click(screen.getByText('parent'));
+
+      // Close menu
+      userEvent.click(screen.getByText('parent'));
+
+      expect(screen.queryByTestId('dropdown-link-menu')).not.toBeInTheDocument();
+    });
+
+    it('Opens / closes on mouse enter and leave', async function () {
+      jest.useFakeTimers();
+
+      render(<NestedDropdown />);
+
+      // Open menu
+      userEvent.click(screen.getByText('parent'));
+
+      fireEvent.mouseEnter(screen.getByText('nested'));
+
+      // Nested menus have delay on open
+      jest.runAllTimers();
+
+      await screen.findByText('nested #2');
+
+      // Leaving Nested Menu
+      fireEvent.mouseLeave(screen.getByText('nested'));
+
+      // Nested menus have close delay
+      jest.advanceTimersByTime(MENU_CLOSE_DELAY - 1);
+
+      // Re-entering nested menu will cancel close
+      fireEvent.mouseEnter(screen.getByText('nested'));
+      jest.advanceTimersByTime(2);
+      expect(screen.getByText('nested #2')).toBeInTheDocument();
+
+      // Re-entering an actor will also cancel close
+      jest.advanceTimersByTime(MENU_CLOSE_DELAY - 1);
+
+      jest.advanceTimersByTime(2);
+      fireEvent.mouseEnter(screen.getByText('parent'));
```

Comment:
shall we use `useEvent` here? 

```suggestion
      userEvent.hover(screen.getByText('parent'));
```

Reply:
Ah yes thank you, gladly will replace fireEvent here